inclinometer/scripts/IMU_pos_computer.py

```diff
@@ -2,64 +2,58 @@
 
 import rospy
 from sensor_msgs.msg import Imu
-from geometry_msgs.msg import Pose
+from sensor_msgs.msg import JointState
 import numpy as np
 
 class NodePositionComputer:
     def __init__(self):
-        self.subscriber1 = rospy.Subscriber("/imu1/sensordata", Imu, self.Imu1Publish)
-        self.subscriber2 = rospy.Subscriber("/imu2/sensordata", Imu, self.Imu2Publish)
-        self.publisher1 = rospy.Publisher('/yelldozer/inclinometer', Pose, queue_size=10)
-        self.publisher2 = rospy.Publisher('/imu2/inclinometer', Pose, queue_size=10)
-        self.test_publisher = rospy.Publisher('/data_test', Pose, queue_size=10)
-        self.imu_pose_msg = [Pose() for i in range(3)]
-        self.test_msg = Pose()
+        self.subscriber1 = rospy.Subscriber("/imu1/sensordata", Imu, self.Imu1Callback)
+        self.subscriber2 = rospy.Subscriber("/imu2/sensordata", Imu, self.Imu2Callback)
+        self.subscriber3 = rospy.Subscriber("/imu3/sensordata", Imu, self.Imu3Callback)
+        self.Joint_pub = rospy.Publisher('/yelldozer/joint_states', JointState, queue_size=10)
+        self.imu_recv_msg = [Imu() for i in range(3)]
+        self.JointState_msg = JointState()
         self.setup()
 
     def setup(self):
-        rospy.Timer(rospy.Duration(1.0 / 10.0), self.diffPublisher)
-
-
-    def Imu1Publish(self, msg):
-        self.sendPoseMsg(msg, 0)
-
-    def Imu2Publish(self, msg):
-        self.sendPoseMsg(msg, 1)
-
-    def sendPoseMsg(self, msg, idx):
-        self.imu_pose_msg[idx].position.x = 0
-        self.imu_pose_msg[idx].position.z = 0
-        self.imu_pose_msg[idx].orientation.w = msg.orientation.w
-        self.imu_pose_msg[idx].orientation.x = msg.orientation.x
-        self.imu_pose_msg[idx].orientation.y = msg.orientation.y
-        self.imu_pose_msg[idx].orientation.z = msg.orientation.z
-        if idx == 0 :
-            self.imu_pose_msg[idx].position.y = -0.3
-            self.publisher1.publish(self.imu_pose_msg[idx])
-        elif idx == 1:
-            self.imu_pose_msg[idx].position.y = 0.3
-            self.publisher2.publish(self.imu_pose_msg[idx]) 
-
-    def diffPublisher(self, event=None):
-        # first imu
-        q0 = self.imu_pose_msg[0].orientation.w
-        q1 = self.imu_pose_msg[0].orientation.x
-        q2 = self.imu_pose_msg[0].orientation.y
-        q3 = self.imu_pose_msg[0].orientation.z
+        self.JointState_msg.name = ["gripper_shaft_joint", "bucket_shaft_joint"]
+        rospy.Timer(rospy.Duration(1.0 / 10.0), self.sendJointPos)
+
+    def Imu1Callback(self, msg):
+        self.imu_recv_msg[0] = msg
+
+    def Imu2Callback(self, msg):
+        self.imu_recv_msg[1] = msg
+
+    def Imu3Callback(self, msg):
+        self.imu_recv_msg[2] = msg
 
-        imu1_angle = np.arctan2(2*(q0*q1 + q2*q3), (1 - 2*(q1*q1 + q2*q2))) * 180 / np.pi
+    def getAngle(self, idx):
```

Comment:
Which angle? Roll or pitch? If it is possible, it might be good to have a short comment with the link to the corresponded math here.